utilities/oadp.py

```diff
@@ -0,0 +1,145 @@
+            teardown=teardown,
+            yaml_file=yaml_file,
+            excluded_resources=excluded_resources,
+            storage_location=storage_location,
+            snapshot_move_data=snapshot_move_data,
+            **kwargs,
+        )
+        self.wait_complete = wait_complete
+        self.timeout = timeout
+
+    def __enter__(self):
+        super().__enter__()
+        if self.wait_complete:
+            self.wait_for_status(
+                status=self.Status.COMPLETED,
+                timeout=self.timeout,
+            )
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        delete_velero_resource(resource=self, client=self.client)
+        super().__exit__(exception_type, exception_value, traceback)
+
+
+@contextmanager
+def create_rhel_vm(
+    storage_class,
+    namespace,
+    dv_name,
+    vm_name,
+    rhel_image,
+    client=None,
+    wait_running=True,
+    volume_mode=None,
+):
+    artifactory_secret = get_artifactory_secret(namespace=namespace)
+    artifactory_config_map = get_artifactory_config_map(namespace=namespace)
+    try:
+        dv = DataVolume(
+            name=dv_name,
+            namespace=namespace,
+            source="http",
+            url=get_http_image_url(
+                image_directory=Images.Rhel.DIR,
+                image_name=rhel_image,
+            ),
+            storage_class=storage_class,
+            size=Images.Rhel.DEFAULT_DV_SIZE,
+            api_name="storage",
+            volume_mode=volume_mode,
+            secret=artifactory_secret,
+            cert_configmap=artifactory_config_map.name,
+        )
+        dv.to_dict()
+        dv_metadata = dv.res["metadata"]
+        with VirtualMachineForTests(
+            client=client,
+            name=vm_name,
+            namespace=dv_metadata["namespace"],
+            os_flavor=OS_FLAVOR_RHEL,
+            memory_guest=Images.Rhel.DEFAULT_MEMORY_SIZE,
+            data_volume_template={"metadata": dv_metadata, "spec": dv.res["spec"]},
+            run_strategy=VirtualMachine.RunStrategy.ALWAYS,
+        ) as vm:
+            if wait_running:
+                running_vm(vm=vm, wait_for_interfaces=True)
+            yield vm
+    finally:
+        cleanup_artifactory_secret_and_config_map(
+            artifactory_secret=artifactory_secret, artifactory_config_map=artifactory_config_map
+        )
+
+
+def check_file_in_vm(vm):
+    with console.Console(vm=vm) as vm_console:
+        vm_console.sendline(LS_COMMAND)
+        vm_console.expect(FILE_NAME_FOR_BACKUP, timeout=TIMEOUT_20SEC)
+        vm_console.sendline(f"cat {FILE_NAME_FOR_BACKUP}")
+        vm_console.expect(TEXT_TO_TEST, timeout=TIMEOUT_20SEC)
+
+
+def is_storage_class_support_volume_mode(storage_class_name, requested_volume_mode):
+    for claim_property_set in StorageProfile(name=storage_class_name).claim_property_sets:
+        if claim_property_set.volumeMode == requested_volume_mode:
```

Comment:
`claim_property_sets` is a list, no? It's not failing on `claim_property_set.volumeMode`?